### DISCUSSION TITLE
Problem with BenchmarksDotNet

### DISCUSSION THREAD

**Discussion by DeafMan1983** [dotnet/Silk.NET#519] [Q&A] [answered]
# Summary
I have problem because Rider said error popup message about BenchmarksDotNet under Linux.

# Steps to reproduce
- Platform: Desktop - Ubuntu 20.04
- Framework Version: .NET Core 3.1.400, .NET 5.0.201,5.0.301
- API: Vulkan 1.2 /OpenGL 4.6
- API Version: Vulkan 1.2/OpenGL 4.5 Core/OpenGLES 3.2
1. It happens It cannot upgrade if I use Rider and It throws error. But I don't know why does it happen with after ./build.sh and hangs longer
2. Use Rider to build and it throws error because BenchmarksDotNet has support _4.8 - But I don't have DotNet 4.8 - But I already installed Mono runtime ( latest version )

# Comments
Can't build with Silk.Net

**Comment by HurricanKai** [comment]
For your purposes, just remove the target framework in there.
But you generally don't need to build the benchmarks if you don't want to run them. So unloading the project in Rider should also work.

**Comment by DeafMan1983** [comment]
Thanks and where is `CreateDefaultContext`?
Like GetApi() from GLFW, SDL2, OpenGl and any .... But I can't find there.  

Ok I will unload it. Thanks!

**Comment by HurricanKai** [comment]
CreateDefaultContext is generated by our source generator. It will be marked as an error in editor, but at compile time it works.

**Comment by DeafMan1983** [comment]
Thank you for explanation! Now I close it.

**Comment by DeafMan1983** [comment]
Now I have problem with CreateDefaultContext
![image](https://user-images.githubusercontent.com/57066679/121657759-f9d50f00-caa0-11eb-9df6-cb395d7b7d2d.png)

I have tried to generate with X11 gen but it looks not working - How do I get success but why does Rider not compile successfully? But for me always errors... I don't understand .... How did you make like magic? I have tried `public partial (type) implementname(....)`
![image](https://user-images.githubusercontent.com/57066679/121658358-87186380-caa1-11eb-891a-18b3d0c4fa0f.png)
But nothing works for me. :(

**Comment by HurricanKai** [comment]
A relatively new .NET SDK is required for the projects to compile. What SDK are you using

**Comment by DeafMan1983** [comment]
I am using Net 5.0 SDK 5.0.301 and Mono 6.12.0.122 
Full detail of DotNet:
```
.NET SDK (reflecting any global.json):
 Version:   5.0.301
 Commit:    cc8c0abb32

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /usr/share/dotnet/sdk/5.0.301/

Host (useful for support):
  Version: 5.0.7
  Commit:  556582d964

.NET SDKs installed:
  2.1.816 [/usr/share/dotnet/sdk]
  3.1.410 [/usr/share/dotnet/sdk]
  5.0.201 [/usr/share/dotnet/sdk]
  5.0.301 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.28 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.28 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.16 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.7 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.28 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.16 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by HurricanKai** [comment]
I'm not sure how mono influences this. Do make sure you have the net5 sdk selected in the rider settings

**Comment by DeafMan1983** [comment]
Still using Net 5.0 Sdk.
Found Rider's Setting...
![image](https://user-images.githubusercontent.com/57066679/121666729-dc587300-caa9-11eb-9435-3d7231d92a67.png)


![image](https://user-images.githubusercontent.com/57066679/121664780-250f2c80-caa8-11eb-9951-21ff898ef0dd.png)

![image](https://user-images.githubusercontent.com/57066679/121665798-df069880-caa8-11eb-993d-765eea2f41a9.png)

**Comment by DeafMan1983** [comment]
@HurricanKai 
Please clean up and use current version "net5.0" for all projects of Silk.NET! Remove all unused frameworks! Thanks!
I really hate problem with errors. I have tried no success. I am very disappointed because it can't make my own X11 for Silk.NET

**Comment by HurricanKai** [comment]
If you can check in any changes you made and link to them here, I'll have a look

**Comment by Perksey** [comment]
> Please clean up and use current version "net5.0" for all projects of Silk.NET! Remove all unused frameworks!

.NET 5 can't be used on mobile. We intend to move to .NET 6 only for Silk.NET 3.0.

> I have tried no success. I am very disappointed because it can't make my own X11 for Silk.NET

You've installed the SilkTouch NuGet package right?

---

Also, please stop opening issues. Issues are not the place for questions or discussions like these, discussions are. Going forward please only open an issue once we've confirmed it's a bug via GitHub Discussions.

**Reply by DeafMan1983** [reply]
`.NET 5 can't be used on mobile. We intend to move to .NET 6 only for Silk.NET 3.0.`
Oh I see that is my bad. Thanks!

I am sorry for that. I remember that. I need know to create discussions

So now I am okay - I am very busy with X11 like embedding on SDL2/GLFW or ModalWindow with System-Engine bridger like I use under Linux then X11 and Wayland for ModalWindow or MultipleWindow 

I saw GLFW3 has not ModalWindow or MultipleWindow support. I thought like helper of SDL2 and GLFW3 made alive.
For Windows Users are happy because they know SetParent and SetWindowPos with SDL2 or GLFW3

For Linux Users use with X11 or Wayland via XReparentWindow or XSetWMTransientFor or other special methods
For macOS Users use with Cocoa or AppKit via Panel or PopupPanel...
For Android Users use with PopupDialog ( Xamarin.Android )
and more...

Do you like my idea? I hope advanced mode of Windowing Engine with GUI. Example "GameMenu" => example Quake Game Menu or "GUIinContext" => CrazyBump or ModelViewer.

Best wishes!

**Comment by DeafMan1983** [comment, marked as answer]
Huhu my friends, @HurricanKai @Perksey and other c# coders,
I would like to show you nice example with Tutorial 2 OpenGL with Silk.NET with X11's subwindow
Woohoo! I made nice Windowing.X11 on Silk.NET for Linux, Unix and FreeBSD
Look out on [Youtube](https://youtu.be/TDfSLR3cZCk) 
Do you believe that?

// UPDATE:
Yay I am very excited to draw rectangle and resize fine. like that.
XDrawable.DrawRectangle() void.
![image](https://user-images.githubusercontent.com/57066679/121774623-835d0d80-cb83-11eb-97cc-bc1c307b8a3b.png)
![image](https://user-images.githubusercontent.com/57066679/121774632-9374ed00-cb83-11eb-9581-282f0329583d.png)

It looks awesome. I continue again my work... PS: My eyes are bit hurting. I need take rest and work later my development.

Stay safe as well, my friends!

**Reply by Perksey** [reply]
Fair enough, well done.
